src/dsp_tools/commands/excel2json/properties.py

```diff
@@ -226,32 +229,33 @@ def _get_final_series(
     return final_series
 
 
-def _row2prop(df_row: pd.Series[Any], row_num: int, excelfile: str) -> dict[str, Any]:
-    _property = {x: df_row[x] for x in mandatory_properties} | {
-        "labels": get_labels(df_row=df_row),
-        "super": [s.strip() for s in df_row["super"].split(",")],
-    }
-    if not pd.isna(df_row["subject"]):
-        _property["subject"] = df_row["subject"]
-
+def _row2prop(df_row: pd.Series[Any], row_num: int, excelfile: str) -> Property:
+    subj = df_row["subject"] if not pd.isna(df_row["subject"]) else None
+    comment = get_comments(df_row=df_row)
     gui_attrib = _get_gui_attribute(df_row=df_row, row_num=row_num)
     match gui_attrib:
-        case dict():
-            _property["gui_attributes"] = gui_attrib
         case InvalidExcelContentProblem():
             msg = f"There is a problem with the excel file: '{excelfile}'\n" + gui_attrib.execute_error_protocol()
             raise InputError(msg) from None
-
-    if comment := get_comments(df_row=df_row):
-        _property["comments"] = comment
-
-    return _property
+        case _:
+            pass
+    prop = Property(
+        name=df_row["name"],
+        super=[s.strip() for s in df_row["super"].split(",")],
+        object=df_row["object"],
+        subject=subj,
+        labels=get_labels(df_row=df_row),
+        comments=comment,
+        gui_element=df_row["gui_element"],
+        gui_attributes=gui_attrib,
+    )
+    return prop
 
 
 def _get_gui_attribute(
     df_row: pd.Series[Any],
     row_num: int,
-) -> dict[str, int | str | float] | InvalidExcelContentProblem | None:
+) -> GuiAttributes | InvalidExcelContentProblem | None:
```

Comment:
I wanted to change the serialisation because in future I would like to do all the invalid checks during the construction, this way we can collect all the errors and communicate them at the end, right now the program stops if one error was found. Also better typing.

Reply:
The type annotations look much cleaner now, thanks for this improvement 👍